### ISSUE TITLE
Levoit Air Purifyer 300s

### ISSUE THREAD

**Issue by icebear74** [webdjoe/pyvesync#92]
Hi,
can you include this one ? What you need for it ?

Tnx from Germany ..
Holger

**Comment by jparchem** [comment]
This is already supported in the library. 

If you are referring to support in Home Assistant, I have a PR submitted that would also expose it there too.  It’s awaiting reviews and approvals.

https://github.com/home-assistant/core/pull/57126

**Comment by icebear74** [comment]
Hi,
it seems, there are Bugs ....

test.py
from pyvesync import VeSync

manager = VeSync("xx@yy.com", "xxx")
manager.login()

# Get/Update Devices from server - populate device lists
manager.get_devices()
#manager.update()
manager.update_all_devices()
# Display outlet device information
for device in manager.fans:
    device.display()


Output:
root@NVidia:~# python3 test.py
Traceback (most recent call last):
  File "test.py", line 9, in <module>
    manager.update_all_devices()
  File "/usr/local/lib/python3.6/dist-packages/pyvesync/vesync.py", line 307, in update_all_devices
    dev.get_details()
AttributeError: 'str' object has no attribute 'get_details'

Same PY but with only update (not update_all_devices)
Device Name:... Luftreiniger
Model: ........ Core300S
Subdevice No: . None
Status: ....... on
Online: ....... online
Type: ......... wifi-air
CID ...
UUID ....
Mode: .............. Mode:  Mode:
Filter Life: ....... Filter Life:  Filter Life:
Fan Level: ......... Fan Level:  Fan Level:
Display: ........... Display:  Display:
Child Lock: ........ Child Lock:  Child Lock:
Night Light: ....... Night Light:  Night Light:
Display Config: .... Display Config:  Display Config:
Display_Forever Config:  Display_Forever Config:  Display_Forever Config:

(no Information, output without CID, UUID ....)
I Have 3 300S Purifyer to test ...

**Comment by icebear74** [comment]
And it will try to use it in FHEM :)

**Comment by kitprice** [comment]
I can confirm that the Core300S is like the Core400S.  I changed the source for the device class of the Core300S from VeSyncAir200S300S to VeSyncAir400S and it now works:

Device Name:... Master Bedroom
Model: ........ Core300S
Subdevice No: . None
Status: ....... on
Online: ....... online
Type: ......... wifi-air
CID: .......... XXX
UUID: ......... XXX
Mode: .............. sleep
Filter Life: ....... 99 percent
Air Quality: ....... 1 ug/m3
Fan Level: ......... 1
Display: ........... False
Child Lock: ........ False
Night Light: ....... off
Display Config: .... True
Display_Forever Config:  True

I believe I tried each Core400S method with the Core300S and everything worked (except for the night light).  However, I don't think there is a night light on the Core400S either, but I don't have one yet.
